x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/new_terms/create_new_terms_alert_type.ts

```diff
@@ -82,6 +83,18 @@ export const createNewTermsAlertType = (
           }
           return validated;
         },
+        /**
+         * validate rule params when rule is bulk edited (update and created in future as well)
+         * returned params can be modified (useful in case of version increment)
+         * @param mutatedRuleParams
+         * @returns mutatedRuleParams
+         */
+        validateMutatedParams: (mutatedRuleParams) => {
+          validateImmutable(mutatedRuleParams.immutable);
+          validateIndexPatterns(mutatedRuleParams.index);
```

Comment:
@marshallmain Are you planning to address https://github.com/elastic/kibana/issues/137029#issuecomment-1193792894 in a separate PR? I could do it as well if you want.

Reply:
I was thinking that would be done separately since the data view validation doesn't exist for any rule type yet, right? I intended for this PR to bring the new terms rule type to parity with the other rule types by adding the immutable and index pattern validation.